jaxrs-tck/src/main/java/jakarta/ws/rs/tck/ee/rs/container/requestcontext/security/JAXRSClientIT.java

```diff
@@ -79,7 +76,7 @@ public static WebArchive createDeployment() throws IOException {
 
     WebArchive archive = ShrinkWrap.create(WebArchive.class, "jaxrs_ee_rs_container_requestcontext_security_web.war");
     archive.addClasses(TSAppConfig.class, Resource.class, RequestFilter.class);
-    archive.addAsResource("jakarta/ws/rs/tck/ee/rs/container/requestcontext/security/jaxrs_ee_rs_container_requestcontext_security_web.war.sun-web.xml");
+    archive.addAsWebInfResource("jakarta/ws/rs/tck/ee/rs/container/requestcontext/security/jaxrs_ee_rs_container_requestcontext_security_web.war.sun-web.xml", "sun-web.xml");
```

Comment:
We should indeed go to a situation where there's a porting kit for GlassFish just as other servers, maybe that moment is now.

(p,s. I think the TCK does not or should not test @RolesAllowed at this moment, since it's not defined for Jakarta REST.)
(p.p.s most usages of sun-web.xml are somewhat questionably. One could just as well create the required users with their groups and passwords in one go)

Reply:
@andymc12 My idea is that a property file could be defined for the TCK, and the file will be a set of tripplets:
<test_name><vendor_specific_deployment_descriptor><descriptor_name_in_web_xml>. The test would grab the property file, and add the deployment descriptor to the WAR accordingly. Would that work for IBM/JBoss?

Reply:
test_name is string
vendor_specific_deployment_descriptor is a path to a file
descriptor_name_in_war is string file name

Reply:
https://github.com/eclipse-ee4j/jaxrs-api/issues/1039 was created to track the work for implementation-specific deployment descriptors but no progress was made yet. I will add this suggestion to the issue too.